### ISSUE TITLE
[Bug] Fenix crashes at sign out after the replacement of Fennec

### ISSUE THREAD

**Issue by ebalazs-sv** [mozilla-mobile/android-components#4893]
## Preconditions:
Have Firefox Release installed.
## Steps to reproduce
1. Replace Fennec with Fenix using a build from https://github.com/mozilla-mobile/fenix/wiki/Fennec-Migration;
2. Go to Settings;
3. Tap the "Turn on Sync" button;
4. Tap "Use email instead" button;
5. Insert valid credentials and tap the Sign in button;
### Expected behavior
The account section is updated showing the user's email.
### Actual behavior
Fenix crashes.
### Notes:
The account is connected after the crash.
Reproducible also when signing out.
On Android Q, the behavior is different: There is a continuous loop and a Working state on the sign in page, but the account is connected.
[Logcat](https://github.com/mozilla-mobile/android-components/files/3788115/crash.txt)

### Device information

* Android device: Samsung Galaxy S8 (Android 9) - ARM; Google Pixel (Android Q) - ARM64
* Fenix version: Nightly 191030 GV 71

**Comment by ebalazs-sv** [comment]
Also, on Samsung Galaxy S8 (Android 9), Bookmarks and History are populated only after the crash.

**Comment by ebalazs-sv** [comment]
Hi!
I tested this on Nightly 191102 GV 71 with Samsung Galaxy S8 (Android 9) and the app keeps stopping after the crash, even in stand by mode. Fenix is usable after Clearing data.
[crash](https://github.com/mozilla-mobile/android-components/files/3813430/crash2.txt)

**Comment by ebalazs-sv** [comment]
Hi! This is reproducible on Fennec migration build 2 with Samsung Galaxy S8 (Android 9), at sign out.

**Comment by pocmo** [comment]
Moving this over to our Nightly column to triage.

**Comment by pocmo** [comment]
I tried to reproduce this (Pixel 3, Android 10). It didn't crash. But when I tried to login again the app just froze while loading the login page and I couldn't do anything. :(

**Comment by pocmo** [comment]
https://sentry.prod.mozaws.net/operations/fenix-fennec-nightly/issues/7091294/

```
RecordNotFoundError: Error(

No record for uaid:chid "[..]")
    at mozilla.appservices.push.RustError.intoException(RustError.kt:6)
    at mozilla.appservices.push.PushManager.decrypt(PushManager.kt:16)
    at mozilla.components.feature.push.RustPushConnection.decrypt(Connection.kt:3)
    at mozilla.components.feature.push.AutoPushFeature$onMessageReceived$1.invokeSuspend(AutoPushFeature.kt:15)
    at mozilla.components.feature.push.AutoPushFeature$onMessageReceived$1.invoke
...
(6 additional frame(s) were not displayed)

Error(

No record for uaid:chid "[..]":"[..]")
```

**Comment by jonalmeida** [comment]
The opening issue for this reads like it's related to the push keys missing and not the `RecordNotFoundError`.

This one should be closed, but I'm going to re-open the linked issue since that would be more accurate.

**Comment by jonalmeida** [comment]
This is a duplicate of the Fenix issue: https://github.com/mozilla-mobile/fenix/issues/6663

**Comment by st3fan** [comment]
@ebalazs-sv I think this was fixed in an earlier build. Can you please verify?

**Comment by ebalazs-sv** [comment]
This is not reproducible anymore in the latest Nightly migration build 7 and build 8.
I will close this. Thanks!
